retrieval/discovery/marathon/conversion.go

```diff
@@ -0,0 +1,76 @@
+package marathon
+
+import (
+	"fmt"
+	"strings"
+
+	clientmodel "github.com/prometheus/client_golang/model"
+
+	"github.com/prometheus/prometheus/config"
+)
+
+// AppsToTargetGroups takes an array of Marathon apps and converts them into target groups.
+func AppsToTargetGroups(apps *AppList) map[string]*config.TargetGroup {
+	tgroups := map[string]*config.TargetGroup{}
+	for _, a := range apps.Apps {
+		if isValidApp(&a) {
+			group := createTargetGroup(&a)
+			tgroups[group.Source] = group
+		}
+	}
+	return tgroups
+}
+
+func createTargetGroup(app *App) *config.TargetGroup {
+	var (
+		targets = targetsForApp(app)
+		source  = targetGroupName(app)
+		job     = clientmodel.LabelValue(sanitizeName(app.ID))
+		image   = clientmodel.LabelValue(imageName(app))
```

Comment:
Especially as our relabelling doesn't support matching source labels via regexes and we would have to come up with some nasty encoding ;)